content/docs/pipelines/sdk/dsl-overview.md

```diff
@@ -0,0 +1,118 @@
++++
+title = "DSL Overview"
+description = "Introduction to the Kubeflow Pipelines domain-specific language (DSL)"
+weight = 20
++++
+
+The
+[Kubeflow Pipelines DSL](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/dsl)
+is a set of Python libraries that you can use to specify machine learning (ML)
+workflows, including pipelines and their components. (If you're new to
+pipelines, see the conceptual guides to [pipelines](/docs/pipelines/concepts/pipeline/)
+and [components](/docs/pipelines/concepts/component/).)
+
+The
+[DSL compiler](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/compiler)
+compiles your Python DSL code into a single static configuration (YAML)
+that the Pipeline Service can process. The Pipeline Service, in turn, converts
+the static configuration into a set of Kubernetes resources for execution.
+
+## Installing the DSL
+
+The DSL is part of the Kubeflow Pipelines software development kit (SDK),
+which includes the DSL as well as Python libraries to interact with the Kubeflow
+Pipeline APIs.
+
+Follow the guide to 
+[installing the Kubeflow Pipelines SDK](/docs/pipelines/sdk/install-sdk/).
+
+## Introduction to main DSL functions and classes
+
+This section introduces the DSL functions and classes that you use most often.
+You can see all classes and functions in the
+[Kubeflow Pipelines DSL](https://github.com/kubeflow/pipelines/tree/master/sdk/python/kfp/dsl).
+
+### Pipelines
+
+To create a pipeline, write your own pipeline function and use the DSL's
+[`pipeline(name, description)` function](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_pipeline.py)
+as a decorator.
+
+Usage:
+
+```python
+@kfp.dsl.pipeline(
+  name='My pipeline',
+  description='My machine learning pipeline'
+)
+def my_pipeline(a: PipelineParam, b: PipelineParam):
+  ...
+```
+
+**Note:** The
+[`Pipeline()` class](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_pipeline.py)
+is not useful for creating pipelines. Instead, you should define your pipeline
+function and decorate it with `@kfp.dsl.pipeline` as described above. The class
+is useful for getting a pipeline object and its operations when implementing a 
+compiler.
+
+### Components
+
+To create a component for your pipeline, write your own component function and
```

Comment:
Could you specify that the component decorator requires the function to return a ContainerOp instance? And, the main purpose is for DSL static type checking.

Reply:
Done